pkg/network/setup/netstat.go

```diff
@@ -139,68 +144,64 @@ func vmiInterfaceKey(vmiUID types.UID, interfaceName string) string {
 	return fmt.Sprintf("%s/%s", vmiUID, interfaceName)
 }
 
-func ifacesStatusFromDomain(vmiStatusIfaces []v1.VirtualMachineInstanceNetworkInterface, domainSpecIfaces []api.Interface) []v1.VirtualMachineInstanceNetworkInterface {
+func ifacesStatusFromDomain(domainSpecIfaces []api.Interface) []v1.VirtualMachineInstanceNetworkInterface {
+	var vmiStatusIfaces []v1.VirtualMachineInstanceNetworkInterface
+
 	for _, domainSpecInterface := range domainSpecIfaces {
 		interfaceMAC := domainSpecInterface.MAC.MAC
 		domainIfaceAlias := domainSpecInterface.Alias.GetName()
 
-		if interfaceStatus := netvmispec.LookupInterfaceStatusByName(vmiStatusIfaces, domainIfaceAlias); interfaceStatus != nil {
-			interfaceStatus.MAC = interfaceMAC
-			interfaceStatus.InfoSource = netvmispec.InfoSourceDomain
-		} else {
-			networkInterface := v1.VirtualMachineInstanceNetworkInterface{
-				Name:       domainIfaceAlias,
-				MAC:        interfaceMAC,
-				InfoSource: netvmispec.InfoSourceDomain,
-			}
-			vmiStatusIfaces = append(vmiStatusIfaces, networkInterface)
+		networkInterface := v1.VirtualMachineInstanceNetworkInterface{
+			Name:       domainIfaceAlias,
+			MAC:        interfaceMAC,
+			InfoSource: netvmispec.InfoSourceDomain,
 		}
+		vmiStatusIfaces = append(vmiStatusIfaces, networkInterface)
 	}
 	return vmiStatusIfaces
 }
 
 func ifacesStatusFromGuestAgent(vmiIfacesStatus []v1.VirtualMachineInstanceNetworkInterface, vmiIfacesSpec []v1.Interface, guestAgentInterfaces []api.InterfaceStatus) []v1.VirtualMachineInstanceNetworkInterface {
-	vmiInterfacesSpecByName := netvmispec.IndexInterfaceSpecByName(vmiIfacesSpec)
 	vmiInterfacesSpecByMac := netvmispec.IndexInterfaceSpecByMac(vmiIfacesSpec)
 
 	for _, guestAgentInterface := range guestAgentInterfaces {
 		if vmiIfaceStatus := netvmispec.LookupInterfaceStatusByMac(vmiIfacesStatus, guestAgentInterface.Mac); vmiIfaceStatus != nil {
-			updateVMIIfaceStatusWithGuestAgentData(vmiIfaceStatus, guestAgentInterface, vmiInterfacesSpecByName)
+			updateVMIIfaceStatusWithGuestAgentData(vmiIfaceStatus, guestAgentInterface)
+			// For backward compatability with older virt-launchers, apply this logic:
+			// - When the domain status `InterfaceName` field is set, the data originates from the guest-agent.
+			//   This is true for old virt-launchers and new ones alike.
+			// - When the domain status `InterfaceName` field is not set (empty), the data originates from a merge
+			//   at an old virt-launcher and comes from the domain spec. In such cases, no action is to be taken.
+			if guestAgentInterface.InterfaceName != "" {
+				vmiIfaceStatus.InfoSource = netvmispec.InfoSourceDomainAndGA
```

Comment:
worth to have it in a function ?
we might remove it in some point in the future, so it will be self contained

Reply:
The only thing that we may remove in the future is the condition. The comment is explaining it, not the assignment.
So if you we move it to a function, it has a potential to just wrap the one-line assignment, which is very odd.

But I usually judge this per the proposed name. I do not have a good name that can embed in it the condition and assignment for this specific case. If you have a good name to describe it, and not change this name when this future you mentioned will come, then we can consider exporting this to a function.

Reply:
what about `if ! isOutdatedVirtLauncher` ?
```
func isOutdatedVirtLauncher(....) bool {
  return guestAgentInterface.InterfaceName == ""
}
```

And then we will be able to just remove the condition and the function (as now, but it is self contained)
maybe it will allow to move the reasoning comment to the function ? (and if we want to be even more explicit, to explain that only the condition can be removed in the future as you explained)

Reply:
I'm fine to export the condition with the comment to a function.
@AlonaKaplan , are you ok with this?

Reply:
Done